run.go

```diff
@@ -144,21 +143,15 @@ type RunOptions struct {
 	DropCapabilities []string
 	// Devices are the additional devices to add to the containers
 	Devices define.ContainerDevices
-	// Secrets are the available secrets to use in a RUN
-	Secrets map[string]define.Secret
```

Comment:
Not sure but I think this is breaking the public API for `RunOptions` instead of removing could we mark these fields as deprecated instead. @nalind Could confirm this.

Reply:
Let me know if it does break the public API! If it does we can just set `RunOptions.RunMountSources.Secrets = RunOptions.Secrets` or something like that?

Reply:
@nalind What's the deprecation protocol? Do we set `RunOptions.RunMountSources.Secrets = RunOptions.Secrets` in `func (b *Builder) Run(command []string, options RunOptions)` in run_linux.go, or do we mark as deprecated and simply ignore the field?

Reply:
How does the caller benefit from our moving a handful of fields into a nested structure?  Is the disruption to any code that calls our `Run()` method worth it?  Would gathering a subset of `RunOptions` into a `RunMountSources` to pass down to `setupMounts()` right before we call it be worse?

Reply:
> Would gathering a subset of `RunOptions` into a `RunMountSources` to pass down to `setupMounts()` right before we call it be worse?
I could do that instead! So we leave the SSH, Secrets, etc. in their original place in RunOptions, and inside `Run(command []string, options RunOptions)`, just put those into a RunMountSources struct and pass that into setupMounts()

I'll fix that up and re-push :) Thanks!